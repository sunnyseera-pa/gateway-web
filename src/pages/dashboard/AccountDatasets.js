import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { Col, Row, Button, Tabs, Tab, Alert } from 'react-bootstrap';
import googleAnalytics from '../../tracking';
import _ from 'lodash';
import Loading from '../commonComponents/Loading';
import DatasetCard from '../commonComponents/DatasetCard';
import NotFound from '../commonComponents/NotFound';
import SVGIcon from '../../images/SVGIcon';
import utils from '../../utils/DataSetHelper.util';
import './Dashboard.scss';
import AccountContent from './Components/AccountContent';

var baseURL = require('../commonComponents/BaseURL').getURL();

const AccountDatasets = props => {
	const [key, setKey] = useState(!_.isEmpty(props.alert.tab) ? props.alert.tab : 'active');
	const [datasetList, setDatasetList] = useState([]);
	const [isLoading, setIsLoading] = useState(true);
	const [activeCount, setActiveCount] = useState(0);
	const [reviewCount, setReviewCount] = useState(0);
	const [archiveCount, setArchiveCount] = useState(0);
	const [rejectedCount, setRejectedCount] = useState(0);
	const [alert] = useState(props.alert);
	const [team, setTeam] = useState(props.team);
	const [publisherID, setPublisherID] = useState('');

	useEffect(() => {
		setTeam(props.team);
	}, [props]);

	useEffect(() => {
		setIsLoading(true);
		doDatasetsCall();
	}, [team]);

	const doDatasetsCall = async () => {
		setIsLoading(true);
		let isPublisher = utils.getPublisherID(props.userState[0], props.team);
		setPublisherID(isPublisher);

		await axios.get(baseURL + `/api/v1/dataset-onboarding/publisher/${isPublisher}`).then(res => {
			if (_.has(res, 'data.data.listOfDatasets')) {
				setDatasetList(res.data.data.listOfDatasets);

				let activeCount = 0;
				let reviewCount = 0;
				let archiveCount = 0;
				let rejectedCount = 0;

				res.data.data.listOfDatasets.forEach(dataset => {
					if (dataset.activeflag === 'active' || dataset.activeflag === 'draft') activeCount++;
					else if (dataset.activeflag === 'inReview') reviewCount++;
					else if (dataset.activeflag === 'archive') archiveCount++;
					else if (dataset.activeflag === 'rejected') rejectedCount++;
				});

				setActiveCount(activeCount);
				setReviewCount(reviewCount);
				setArchiveCount(archiveCount);
				setRejectedCount(rejectedCount);
			}
			if (isPublisher === 'admin') setKey('inReview');
			setIsLoading(false);
		});
	};

	const createNewDataset = e => {
		e.preventDefault();
		//call to API to create new dataset
		setIsLoading(true);
		let isPublisher = utils.getPublisherID(props.userState[0], props.team);
		axios.post(baseURL + '/api/v1/dataset-onboarding', { publisherID: isPublisher }).then(res => {
			let { id } = res.data.data;
			//load dataset onboarding page
			if (!_.isUndefined(id)) window.location.href = `/dataset-onboarding/${id}`;
			else {
				//show error message
				setIsLoading(false);
			}
		});
	};

	const handleSelect = key => {
		setKey(key);
	};

	const getDatasetPath = id => {
		return `/account/datasets/${id}`;
	};

	const generateAlert = () => {
		let { message = '' } = alert;

		return (
			<AccountContent className='mt-3'>
				<Alert variant={'success'} className='col-sm-12 main-alert'>
					<SVGIcon name='check' width={18} height={18} fill={'#2C8267'} /> {message}
				</Alert>
			</AccountContent>
		);
	};

	if (isLoading) {
		return (
			<AccountContent>
				<Loading />
			</AccountContent>
		);
	}

	return (
		<div>
			<>{!_.isEmpty(alert) ? generateAlert() : ''}</>
			<AccountContent>
				<div className='accountHeader'>
					<Row>
						<Col sm={12} md={8}>
							<div>
								<span className='black-20'>Datasets</span>
							</div>
							<div>
								<span className='gray700-13 '>
									{publisherID !== 'admin'
										? 'View, add, edit, archive and check the status of your datasets.'
										: 'Approve or reject pending datasets'}
								</span>
							</div>
						</Col>
						<Col sm={12} md={4} style={{ textAlign: 'right' }}>
							<Button
								variant='primary'
								className='addButton'
								onClick={
									(() => googleAnalytics.recordEvent('Datasets', 'Add a new dataset', 'Datasets dashboard button clicked'),
									createNewDataset)
								}>
								+ Add a new dataset
							</Button>
						</Col>
					</Row>
				</div>
				<div className='tabsBackground'>
					<Row>
						<Col sm={12} lg={12}>
							{team === 'admin' ? (
								<Tabs className='dataAccessTabs gray700-13' activeKey={key} onSelect={handleSelect}>
									<Tab eventKey='inReview' title={'Pending approval (' + reviewCount + ')'}>
										{' '}
									</Tab>
								</Tabs>
							) : (
								<Tabs className='dataAccessTabs gray700-13' activeKey={key} onSelect={handleSelect}>
									<Tab eventKey='active' title={'Active (' + activeCount + ')'}>
										{' '}
									</Tab>
									<Tab eventKey='inReview' title={'Pending approval (' + reviewCount + ')'}>
										{' '}
									</Tab>
									<Tab eventKey='rejected' title={'Rejected (' + rejectedCount + ')'}>
										{' '}
									</Tab>
									<Tab eventKey='archive' title={'Archived (' + archiveCount + ')'}>
										{' '}
									</Tab>
								</Tabs>
							)}
						</Col>
					</Row>
				</div>

				{(() => {
					switch (key) {
						case 'active':
							return (
								<div>
									{activeCount <= 0 ? (
										<NotFound word='datasets' />
									) : (
										datasetList.map(dataset => {
											if (dataset.activeflag !== 'active' && dataset.activeflag !== 'draft') {
												return <></>;
											} else {
												return (
													<DatasetCard
														id={dataset._id}
														title={dataset.name}
														publisher={dataset.datasetv2.summary.publisher.name}
														version={dataset.datasetVersion}
														isDraft={true}
														datasetStatus={dataset.activeflag}
														timeStamps={dataset.timestamps}
														completion={dataset.percentageCompleted}
														listOfVersions={dataset.listOfVersions}
													/>
												);
											}
										})
									)}
								</div>
							);
						case 'inReview':
							return (
								<div>
									{reviewCount <= 0 ? (
										<NotFound word='datasets' />
									) : (
										datasetList.map(dataset => {
											if (dataset.activeflag !== 'inReview') {
												return <></>;
											} else {
												return (
													<DatasetCard
														path={getDatasetPath(dataset.pid)}
														id={dataset._id}
														title={dataset.name}
														publisher={dataset.datasetv2.summary.publisher.name}
														version={dataset.datasetVersion}
														isDraft={true}
														datasetStatus={dataset.activeflag}
														timeStamps={dataset.timestamps}
														completion={dataset.percentageCompleted}
														listOfVersions={dataset.listOfVersions}
													/>
												);
											}
										})
									)}
								</div>
							);
						case 'rejected':
							return (
								<div>
									{rejectedCount <= 0 ? (
										<NotFound word='datasets' />
									) : (
										datasetList.map(dataset => {
											if (dataset.activeflag !== 'rejected') {
												return <></>;
											} else {
												return (
													<DatasetCard
														id={dataset._id}
														title={dataset.name}
														publisher={dataset.datasetv2.summary.publisher.name}
														version={dataset.datasetVersion}
														isDraft={true}
														datasetStatus={dataset.activeflag}
														timeStamps={dataset.timestamps}
														completion={dataset.percentageCompleted}
														rejectionText={dataset.applicationStatusDesc}
														rejectionAuthor={dataset.applicationStatusAuthor}
														listOfVersions={dataset.listOfVersions}
													/>
												);
											}
										})
									)}
								</div>
							);
						case 'archive':
							return (
								<div>
									{archiveCount <= 0 ? (
										<NotFound word='datasets' />
									) : (
										datasetList.map(dataset => {
											if (dataset.activeflag !== 'archive') {
												return <></>;
											} else {
												return (
													<DatasetCard
														id={dataset._id}
														title={dataset.name}
														//publisher={dataset.datasetv2.summary.publisher.name}
														version={dataset.datasetVersion}
														isDraft={true}
														datasetStatus={dataset.activeflag}
														timeStamps={dataset.timestamps}
														completion={dataset.percentageCompleted}
														listOfVersions={dataset.listOfVersions}
													/>
												);
											}
										})
									)}
								</div>
							);
						default:
							return key;
					}
				})()}
			</AccountContent>
		</div>
	);
};

export default AccountDatasets;
