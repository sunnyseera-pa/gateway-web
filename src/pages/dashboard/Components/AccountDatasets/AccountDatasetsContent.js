import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import pluralize from 'pluralize';
import DatasetCard from '../../../commonComponents/DatasetCard';
import SearchResults from '../../../commonComponents/SearchResults';
import Icon from '../../../../components/Icon';
import SearchControls from '../../../../components/SearchControls';
import '../../Dashboard.scss';
import MessageNotFound from '../../../commonComponents/MessageNotFound';

const AccountDatasetsContent = ({ data = [], onSubmit, isLoading, isFetched, status, params, team, count }) => {
	const [searchValue, setSearchValue] = useState();
	const history = useHistory();

	const { t } = useTranslation();
	const handleChange = React.useCallback(value => {
		setSearchValue(value);
	}, []);

	const getDatasetPath = id => {
		return `/account/datasets/${id}`;
	};

	const handleActivityLogClick = id => {
		history.push(getDatasetPath(id));
	};

	const hasActivityHistory = React.useCallback(
		dataset => {
			return dataset.listOfVersions.length > 0 && team === 'admin';
		},
		[team]
	);

	const getDatasetCardProps = dataset => {
		const datasetCardProps = {};

		if (dataset.activeflag === 'inReview' && hasActivityHistory(dataset)) {
			datasetCardProps.slaProps = {
				icon: <Icon name='eye' role='button' onClick={() => handleActivityLogClick(dataset.pid)} mr={1} />,
			};

			datasetCardProps.path = getDatasetPath(dataset.pid);
		} else if (dataset.activeflag === 'rejected') {
			datasetCardProps.rejectionText = dataset.applicationStatusDesc;
			datasetCardProps.rejectionAuthor = dataset.applicationStatusAuthor;
		}

		return datasetCardProps;
	};

	const { search, sortBy, sortDirection, maxResults } = params;

	return (
		<>
			{isFetched && (
				<SearchControls
					type={t(`dataset.${status}`)}
					mt={3}
					onSubmit={onSubmit}
					inputProps={{
						onChange: handleChange,
						mt: 2,
						value: search,
					}}
					sortProps={{
						direction: sortDirection,
						value: sortBy,
						options: ['alphabetic', 'latest', 'recentlyadded', 'metadata', 'popularity'],
						mt: 2,
						allowDirection: true,
						minWidth: '300px',
					}}
				/>
			)}

			<SearchResults
				data={data}
				errorMessage={({ type }) => <MessageNotFound word={pluralize(type)} />}
				results={data =>
					data.map(dataset => (
						<DatasetCard
							key={dataset._id}
							id={dataset._id}
							title={dataset.name}
							publisher={status !== 'archive' && dataset.datasetv2.summary.publisher.name}
							version={dataset.datasetVersion}
							isDraft={true}
							datasetStatus={dataset.activeflag}
							timeStamps={dataset.timestamps}
							completion={dataset.percentageCompleted}
							listOfVersions={dataset.listOfVersions}
							{...getDatasetCardProps(dataset)}
						/>
					))
				}
				count={count}
				maxResults={maxResults}
				type='dataset'
				isLoading={isLoading}
				search={searchValue}
			/>
		</>
	);
};

AccountDatasetsContent.defaultProps = {
	params: {
		search: '',
		sortBy: 'latest',
		sortDirection: 'desc',
	},
};

export default AccountDatasetsContent;
