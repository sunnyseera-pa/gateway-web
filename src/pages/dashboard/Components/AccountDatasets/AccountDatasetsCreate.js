import React, { useEffect } from 'react';
import _ from 'lodash';
import { Alert, Button, Col, Row } from 'react-bootstrap';
import SVGIcon from '../../../../images/SVGIcon';
import serviceDatasetOnboarding from '../../../../services/dataset-onboarding/dataset-onboarding';
import googleAnalytics from '../../../../tracking';

const AccountDatasetsCreate = props => {
	const {
		publisherID,
		team,
		alert: { message },
	} = props;

	const dataPostDatasetOnboarding = serviceDatasetOnboarding.usePostDatasetOnboarding({ publisherID }, null, {
		enabled: false,
	});

	const createNewDataset = e => {
		e.preventDefault();

		dataPostDatasetOnboarding.refetch();
	};

	useEffect(() => {
		if (dataPostDatasetOnboarding.data) {
			const {
				data: {
					data: {
						data: { id },
					},
				},
			} = dataPostDatasetOnboarding;

			if (!_.isUndefined(id)) window.location.href = `/dataset-onboarding/${id}`;
		}
	}, [dataPostDatasetOnboarding.data]);

	return (
		<>
			{message && (
				<Alert variant={'success'} className='col-sm-12 main-alert'>
					<SVGIcon name='check' width={18} height={18} fill={'#2C8267'} /> {message}
				</Alert>
			)}
			<div className='accountHeader'>
				<Row>
					<Col sm={12} md={8}>
						<div>
							<span className='black-20'>Datasets</span>
						</div>
						<div>
							<span className='gray700-13 '>
								{publisherID !== 'admin'
									? 'View, add, edit, archive and check the status of your datasets.'
									: 'Approve or reject pending datasets'}
							</span>
						</div>
					</Col>
					<Col sm={12} md={4} style={{ textAlign: 'right' }}>
						{team !== 'admin' && (
							<Button
								variant='primary'
								className='addButton'
								onClick={
									(() => googleAnalytics.recordEvent('Datasets', 'Add a new dataset', 'Datasets dashboard button clicked'),
									createNewDataset)
								}>
								+ Add a new dataset
							</Button>
						)}
					</Col>
				</Row>
			</div>
		</>
	);
};

export default AccountDatasetsCreate;
