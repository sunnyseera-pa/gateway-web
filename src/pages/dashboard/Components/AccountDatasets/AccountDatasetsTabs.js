import React from 'react';
import { Col, Row, Tab, Tabs } from 'react-bootstrap';
import '../../Dashboard.scss';

const AccountDatasetsTabs = ({ counts = { inReview: 0, active: 0, rejected: 0, archive: 0 }, team, onSelectTab, activeKey }) => {
	return (
		<div className='tabsBackground mb-0'>
			<Row>
				<Col sm={12} lg={12}>
					{team === 'admin' ? (
						<Tabs className='dataAccessTabs gray700-13' activeKey={activeKey} onSelect={onSelectTab}>
							<Tab eventKey='inReview' title={'Pending approval (' + counts.inReview + ')'}>
								{' '}
							</Tab>
						</Tabs>
					) : (
						<Tabs className='dataAccessTabs gray700-13' activeKey={activeKey} onSelect={onSelectTab}>
							<Tab eventKey='active' title={'Active (' + counts.active + ')'}>
								{' '}
							</Tab>
							<Tab eventKey='inReview' title={'Pending approval (' + counts.inReview + ')'}>
								{' '}
							</Tab>
							<Tab eventKey='rejected' title={'Rejected (' + counts.rejected + ')'}>
								{' '}
							</Tab>
							<Tab eventKey='archive' title={'Archived (' + counts.archive + ')'}>
								{' '}
							</Tab>
						</Tabs>
					)}
				</Col>
			</Row>
		</div>
	);
};

export default AccountDatasetsTabs;
