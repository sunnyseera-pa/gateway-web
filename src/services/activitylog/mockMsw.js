import { rest } from 'msw';
import { apiV2URL } from '../../configs/url.config';

export const mockDatasetActivityLog = {
	success: true,
	logs: [
		{
			version: 'Version 4.0.0',
			datasetVersion: 4,
			meta: {
				dateSubmitted: '2021-09-03T09:51:44.842Z',
				dateCreated: '2021-09-02T15:41:16.392Z',
				applicationStatus: 'rejected',
			},
			events: [],
		},
		{
			version: 'Version 3.0.0',
			datasetVersion: 3,
			meta: {
				dateSubmitted: '2021-09-02T15:38:29.571Z',
				dateCreated: '2021-09-02T15:37:57.159Z',
				applicationStatus: 'archive',
			},
			events: [],
		},
		{
			version: 'Version 2.0.0',
			datasetVersion: 2,
			meta: {
				dateSubmitted: '2021-09-02T14:47:48.056Z',
				dateCreated: '2021-09-02T14:44:08.889Z',
				applicationStatus: 'rejected',
			},
			events: [],
		},
		{
			version: 'Version 1.0.0',
			datasetVersion: 1,
			meta: {
				dateSubmitted: '2021-09-02T13:29:59.320Z',
				dateCreated: '2021-09-02T13:26:05.761Z',
				applicationStatus: 'archive',
			},
			events: [],
		},
	],
};

export const mswPostDatasetActivityLog = rest.post(`${apiV2URL}/activitylog`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(mockDatasetActivityLog));
});

export default [mswPostDatasetActivityLog];
