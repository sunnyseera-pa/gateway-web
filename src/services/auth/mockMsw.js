import { rest } from 'msw';
import { apiURL } from '../../configs/url.config';
import { mockUser } from './mockData';
export const mswGetAuthStatus = rest.get(`${apiURL}/auth/status`, (req, res, ctx) => {
    return res(
        ctx.status(200),
        ctx.json({
            ...mockUser,
        })
    );
});

export default [mswGetAuthStatus];
