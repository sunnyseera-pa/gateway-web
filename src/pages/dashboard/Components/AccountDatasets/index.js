import AccountDatasets from './AccountDatasets';

export default AccountDatasets;
