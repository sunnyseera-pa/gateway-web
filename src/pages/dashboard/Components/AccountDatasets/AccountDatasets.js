import _ from 'lodash';
import React, { useEffect, useState } from 'react';
import { Alert, Button, Col, Row } from 'react-bootstrap';
import { useAuth } from '../../../../context/AuthContext';
import SVGIcon from '../../../../images/SVGIcon';
import serviceDatasetOnboarding from '../../../../services/dataset-onboarding/dataset-onboarding';
import googleAnalytics from '../../../../tracking';
import utils from '../../../../utils/DataSetHelper.util';
import { LayoutContent } from '../../../storybookComponents/Layout';
import useSearch from '../../../storybookComponents/Search/useSearch';
import '../../Dashboard.scss';
import AccountDatasetsContent from './AccountDatasetsContent';
import AccountDatasetsTabs from './AccountDatasetsTabs';

const AccountDatasets = props => {
	const [key, setKey] = useState();
	const [alert] = useState(props.alert);
	const { userState } = useAuth();
	const [publisherID, setPublisherId] = useState();

	const { isLoading, data, getResults } = useSearch(serviceDatasetOnboarding.useGetPublisher(publisherID), {
		params: {
			maxResults: 0,
		},
		count: data => {
			return data.counts[key];
		},
	});

	const dataPostDatasetOnboarding = serviceDatasetOnboarding.usePostDatasetOnboarding({ publisherID }, null, {
		enabled: false,
	});

	const { team } = props;

	useEffect(() => {
		const initialKey = !_.isEmpty(props.alert.tab) ? props.alert.tab : 'active';

		setPublisherId(utils.getPublisherID(userState[0], team));
		setKey(team === 'admin' ? 'inReview' : initialKey);
	}, [team]);

	useEffect(() => {
		if (publisherID && key) {
			getResults({
				status: key,
				page: 1,
			});
		}
	}, [publisherID, key]);

	useEffect(() => {
		if (dataPostDatasetOnboarding.data) {
			const {
				data: {
					data: { id },
				},
			} = dataPostDatasetOnboarding;

			if (!_.isUndefined(id)) window.location.href = `/dataset-onboarding/${id}`;
		}
	}, [dataPostDatasetOnboarding.data]);

	const handleSubmit = React.useCallback(
		({ search, sortBy }) => {
			getResults({
				search,
				sortBy,
				status: key,
				page: 1,
			});
		},
		[key, publisherID]
	);

	const createNewDataset = e => {
		e.preventDefault();

		dataPostDatasetOnboarding.refetch();
	};

	const handleSelect = key => {
		setKey(key);
	};

	const generateAlert = () => {
		let { message = '' } = alert;

		return (
			<LayoutContent className='mt-3'>
				<Alert variant={'success'} className='col-sm-12 main-alert'>
					<SVGIcon name='check' width={18} height={18} fill={'#2C8267'} /> {message}
				</Alert>
			</LayoutContent>
		);
	};

	const { isLoading: isLoadingCreateDataset } = dataPostDatasetOnboarding;

	let statusCounts;
	let datasets;

	if (data) {
		const {
			data: {
				data: { counts, listOfDatasets },
			},
		} = data;

		statusCounts = counts;
		datasets = listOfDatasets;
	}

	return (
		<div>
			<>{!_.isEmpty(alert) ? generateAlert() : ''}</>
			<LayoutContent>
				<div className='accountHeader'>
					<Row>
						<Col sm={12} md={8}>
							<div>
								<span className='black-20'>Datasets</span>
							</div>
							<div>
								<span className='gray700-13 '>
									{publisherID !== 'admin'
										? 'View, add, edit, archive and check the status of your datasets.'
										: 'Approve or reject pending datasets'}
								</span>
							</div>
						</Col>
						<Col sm={12} md={4} style={{ textAlign: 'right' }}>
							{team !== 'admin' && (
								<Button
									variant='primary'
									className='addButton'
									onClick={
										(() => googleAnalytics.recordEvent('Datasets', 'Add a new dataset', 'Datasets dashboard button clicked'),
										createNewDataset)
									}>
									+ Add a new dataset
								</Button>
							)}
						</Col>
					</Row>
				</div>
				<AccountDatasetsTabs counts={statusCounts} onSelectTab={handleSelect} team={team} activeKey={key} />
				<AccountDatasetsContent
					isLoading={isLoading || isLoadingCreateDataset}
					data={datasets}
					onSubmit={handleSubmit}
					team={team}
					status={key}
				/>
			</LayoutContent>
		</div>
	);
};

export default AccountDatasets;
