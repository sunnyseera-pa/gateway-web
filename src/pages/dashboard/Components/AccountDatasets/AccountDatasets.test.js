import React from 'react';
import { render, waitFor, act } from '@testing-library/react';
import AccountDatasets from './AccountDatasets';
import { server } from '../../../../services/mockServer';
import { mockGetPublisher } from '../../../../services/dataset-onboarding/mockMsw';
import '../../../../utils/test.util';

const mockAccountDatasetsTabs = jest.fn();
const mockAccountDatasetsContent = jest.fn();

jest.mock('./AccountDatasetsTabs', () => props => {
	mockAccountDatasetsTabs(props);

	return (
		<button
			onClick={() => {
				props.onSelectTab('inReview');
			}}>
			Select tab
		</button>
	);
});

jest.mock('./AccountDatasetsContent', () => props => {
	mockAccountDatasetsContent(props);

	return (
		<button
			onClick={() => {
				props.onSubmit({
					search: 'dataset',
					sortBy: 'metadataQuality',
				});
			}}>
			Submit
		</button>
	);
});

jest.mock('../../../../components/Icon', () => 'Icon');

const props = {
	alert: { message: 'Message goes here' },
	team: 'applicant',
};

let wrapper;

describe('Given the AccountDatasets component', () => {
	beforeAll(() => {
		server.listen();
	});

	afterEach(() => {
		server.resetHandlers();
	});

	afterAll(() => {
		server.close();
	});

	describe('When it is rendered', () => {
		beforeAll(async () => {
			wrapper = render(<AccountDatasets {...props} />, {
				wrapper: Providers,
			});

			await waitFor(() => expect(wrapper.getByText('Select tab')).toBeTruthy());
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then shows a success message', () => {
			expect(wrapper.getByText('Message goes here')).toBeTruthy();
		});

		it('Then calls the tabs with the correct props', async () => {
			await waitFor(() => {
				expect(mockAccountDatasetsTabs).toHaveBeenCalledLastWithMatch({
					activeKey: 'active',
					counts: { inReview: 19 },
					team: 'applicant',
				});
			});
		});

		it('Then calls the results with the correct props', async () => {
			await waitFor(() => {
				return expect(mockAccountDatasetsContent).toHaveBeenCalledLastWithMatch({
					data: mockGetPublisher.data.results.listOfDatasets,
					isFetched: true,
					isLoading: false,
					status: 'active',
					team: 'applicant',
				});
			});
		});

		describe('And submit is clicked', () => {
			beforeAll(() => {
				mockAccountDatasetsContent.mockClear();

				act(async () => {
					const submit = wrapper.getByText('Submit');

					await fireEvent.click(submit, {
						search: 'dataset',
						sortBy: 'metadataQuality',
					});
				});
			});

			it('Then calls the results with the correct props', async () => {
				await waitFor(() => {
					return expect(mockAccountDatasetsContent).toHaveBeenCalledLastWithMatch({
						isFetched: true,
						isLoading: false,
						status: 'active',
						team: 'applicant',
					});
				});
			});

			describe('And the tab is clicked', () => {
				beforeAll(() => {
					mockAccountDatasetsContent.mockClear();

					act(async () => {
						const tab = wrapper.getByText('Select tab', 'inReview');

						await fireEvent.click(tab);
					});
				});

				it('Then calls the results with the correct props', async () => {
					await waitFor(() => {
						return expect(mockAccountDatasetsContent).toHaveBeenCalledLastWithMatch({
							data: mockGetPublisher.data.results.listOfDatasets,
							isFetched: true,
							isLoading: false,
							status: 'inReview',
							team: 'applicant',
						});
					});
				});
			});
		});
	});
});
