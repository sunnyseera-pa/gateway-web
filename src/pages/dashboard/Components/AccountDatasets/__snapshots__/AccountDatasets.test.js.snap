// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AccountDatasets component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div>
    
    <div
      class="ui-LayoutContent row"
    >
      <div
        class="ui-LayoutContent--left col-1"
      />
      <div
        class="ui-LayoutContent--content col-10"
      >
        <div
          class="accountHeader"
        >
          <div
            class="row"
          >
            <div
              class="col-md-8 col-sm-12"
            >
              <div>
                <span
                  class="black-20"
                >
                  Datasets
                </span>
              </div>
              <div>
                <span
                  class="gray700-13 "
                >
                  View, add, edit, archive and check the status of your datasets.
                </span>
              </div>
            </div>
            <div
              class="col-md-4 col-sm-12"
              style="text-align: right;"
            >
              <button
                class="addButton btn btn-primary"
                type="button"
              >
                + Add a new dataset
              </button>
            </div>
          </div>
        </div>
        <button>
          Select tab
        </button>
        <button>
          Submit
        </button>
      </div>
      <div
        class="ui-LayoutContent--right col-1"
      />
    </div>
  </div>
</div>
`;
